gcc/rust/ast/rust-ast-collector.cc

```diff
@@ -125,22 +144,30 @@ TokenCollector::visit (FunctionParam &param)
 	}
       push (Rust::Token::make (ELLIPSIS, UNDEF_LOCATION));
     }
+
+  end_internal_comment ("FunctionParam");
 }
 
 void
 TokenCollector::visit (VariadicParam &param)
 {
+  begin_internal_comment ("VariadicParam");
+
   if (param.has_pattern ())
     {
       visit (param.get_pattern ());
       push (Rust::Token::make (COLON, UNDEF_LOCATION));
     }
   push (Rust::Token::make (ELLIPSIS, UNDEF_LOCATION));
+
+  end_internal_comment ("VariadicParam");
```

Comment:
This pattern is nice but if forces you to write the node's "name" twice - once when you begin the comment, and once when you end it. It would be nicer and easier to maintain if this was wrapped in a lambda of some sort, e.g.:

```c++
internal_comment("VariadicParam", []() {
      if (param.has_pattern ())
    {
      visit (param.get_pattern ());
      push (Rust::Token::make (COLON, UNDEF_LOCATION));
    }
  push (Rust::Token::make (ELLIPSIS, UNDEF_LOCATION));
  });
  ```
  
  and the code for `internal_comment` could be something like the following:
  
  ```c++
  template <typename T> internal_comment(std::string node_name, std::function<void(T& node)> visitor) {
      begin_internal_comment(node_name);
      
      visitor();
      
      end_internal_comment(node_name);
}
```

note that the ergonomics might not be amazing, in which case we can look at something else like a custom destructor like what @jdupak did recently (we talked about it a little bit on Zulip)